src/Bicep.Core/TypeSystem/Providers/Extensibility/ExtensionResourceTypeFactory.cs

```diff
@@ -245,6 +245,13 @@ private static ResourceFlags ToResourceFlags(Azure.Bicep.Types.Concrete.Resource
                 output |= ResourceFlags.ReadOnly;
             }
 
+            // Resource is WriteOnly if there are no readable scopes but has writable scopes
+            // This is specifically for extensions which cannot be referenced as existing
+            if (readableScopes == ResourceScope.None && writableScopes != ResourceScope.None)
```

Comment:
This reminds me - should we also check if `readableScopes` is not none when adding the read-only flag above (line 243)?